### PR TITLE
The /status table command was getting slower when we had multiple trades opened

### PR DESCRIPTION
**Pull request by jblestang**
For each trades opened, the telegram bot was fetching the last bid price making a request towards bittrex.com. This PR allow the use of cached values retrieved in the _process function and fix issue #278. 

By opportunity all calls to get the bid prices where replaced in the Bot.

### CODE REVIEW

**Comment by glonlas** [comment]
Very nice PR, I like it. Only one comment, there is no unitest to test the new « refresh » logic in get_ticker().
Could you add this unit test to test the function when you refresh the value and when you read it from the cache? Then we are good to go.

**Comment by jblestang** [comment]
Yes sure!

File: freqtrade/exchange/interface.py [hunk 1]
```diff
@@ -62,10 +62,11 @@ def get_balances(self) -> List[dict]:
         """
 
     @abstractmethod
-    def get_ticker(self, pair: str) -> dict:
+    def get_ticker(self, pair: str, refresh: Optional[bool] = True) -> dict:
         """
         Gets ticker for given pair.
         :param pair: Pair as str, format: BTC_ETC
+        :param refresh: Shall we query a new value or a cached value is enought
```
Comment (vertti) [inline freqtrade/exchange/interface.py]: Type enought -> enough.

File: freqtrade/tests/exchange/test_exchange.py [hunk 2]
```diff
@@ -163,7 +168,19 @@ def test_get_ticker(mocker, ticker):
     ticker = get_ticker(pair='BTC_ETH')
     assert ticker['bid'] == 0.00001098
     assert ticker['ask'] == 0.00001099
+
+    # if not caching the result we should get the same ticker
+    ticker = get_ticker(pair='BTC_ETH', refresh=False)
```
Comment (glonlas) [inline freqtrade/tests/exchange/test_exchange.py]: This condition is not tested properly.  You should change the ticker before to see that get_ticker() still return the same.

The test could look like that.
```python
def test_get_ticker(mocker, ticker):

    api_mock = MagicMock()
    api_mock.get_ticker = MagicMock(return_value=ticker())
    mocker.patch('freqtrade.exchange._API', api_mock)

    ticker = get_ticker(pair='BTC_ETH')
    assert ticker['bid'] == 0.00001098
    assert ticker['ask'] == 0.00001099

    # if not caching the result we should get the same ticker
    api_mock.get_ticker = MagicMock(return_value={"bid": 0, "ask": 1})
    mocker.patch('freqtrade.exchange._API', api_mock)
    ticker = get_ticker(pair='BTC_ETH', refresh=False)
    assert ticker['bid'] == 0.00001098
    assert ticker['ask'] == 0.00001099

    # change the ticker
    api_mock.get_ticker = MagicMock(return_value={"bid": 2, "ask": 3})
    mocker.patch('freqtrade.exchange._API', api_mock)

    ticker = get_ticker(pair='BTC_ETH', refresh=True)
    assert ticker['bid'] == 2
    assert ticker['ask'] == 3
```

However, when I execute this test, it fails.

```bash
22:01 $ pytest freqtrade/tests/exchange/
============================= test session starts ==============================
platform darwin -- Python 3.6.3, pytest-3.3.1, py-1.5.2, pluggy-0.6.0
rootdir: /Users/geraldlonlas/Desktop/dev/freqtrade, inifile:
plugins: mock-1.6.3, cov-2.5.1
collected 21 items

freqtrade/tests/exchange/test_exchange.py ..............F...             [ 85%]
freqtrade/tests/exchange/test_exchange_bittrex.py ...                    [100%]

=================================== FAILURES ===================================
_______________________________ test_get_ticker ________________________________

mocker = <pytest_mock.MockFixture object at 0x1073437b8>
ticker = {'ask': 1, 'bid': 0}

    def test_get_ticker(mocker, ticker):

        api_mock = MagicMock()
        api_mock.get_ticker = MagicMock(return_value=ticker())
        mocker.patch('freqtrade.exchange._API', api_mock)

        ticker = get_ticker(pair='BTC_ETH')
        assert ticker['bid'] == 0.00001098
        assert ticker['ask'] == 0.00001099

        # if not caching the result we should get the same ticker
        api_mock.get_ticker = MagicMock(return_value={"bid": 0, "ask": 1})
        ticker = get_ticker(pair='BTC_ETH', refresh=False)
>       assert ticker['bid'] == 0.00001098
E       assert 0 == 1.098e-05

freqtrade/tests/exchange/test_exchange.py:175: AssertionError
===================== 1 failed, 20 passed in 0.19 seconds ======================
```

**Comment by glonlas** [comment]
Any updates on the Unit tests?

**Comment by jblestang** [comment]
Unitest fixed

**Comment by jblestang** [comment]
Sorry I did merge develop instead of rebasing ... (we do merges @ work ... )

**Pull request merged**